include/triton/core/tritonbackend.h

```diff
@@ -595,6 +596,65 @@ TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_ResponseSend(
     TRITONBACKEND_Response* response, const uint32_t send_flags,
     TRITONSERVER_Error* error);
 
+///
+/// TRITONBACKEND_State
+///
+/// Object representing a state.
+///
+
+/// Create a state in the request. The returned state object is only valid
+/// before the TRITONBACKEND_UpdateState is called. The state should not be
+/// freed by the caller. If TRITONBACKEND_UpdateState is not called, the
+/// lifetime of the state matches the lifetime of the request. If the state name
+/// does not exist in the "state" section of the model configuration, the state
+/// will not be updated and an error will be returned.
+///
+/// \param state Returns the new state object.
+/// \param request The request.
+/// \param name The name of the output tensor.
+/// \param datatype The datatype of the output tensor.
+/// \param shape The shape of the output tensor.
+/// \param dims_count The number of dimensions in the output tensor
+/// shape.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_StateNew(
+    TRITONBACKEND_State** state, TRITONBACKEND_Request* request,
+    const char* name, const TRITONSERVER_DataType datatype,
+    const int64_t* shape, const uint32_t dims_count);
+
+/// Update the state for the sequence. Calling this function will replace the
+/// state stored for this seqeunce in Triton with the state provided in the
```

Comment:
...sequence in Triton with 'state'.

You should say something about how it is not required that the backend call this function and if the backend doesn't call this function then the state will not be updated and the next inference request in the sequence will use the existing state... or something like that.

Reply:
Added a sentence about this.